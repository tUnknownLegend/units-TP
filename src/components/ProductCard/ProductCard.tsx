import { Product } from '../../types/Product';
import React, { FC } from 'react';
import { getPrice } from '../../utils';

import './ProductCard.css';

export const ProductCard: FC<Product> = ({
    name,
    description,
    price,
    priceSymbol,
    category,
    imgUrl,
}) => {
    return (
        <div className="product-card">
            <div className="product-card__text">
                <h2 className="product-card__name">{name}</h2>
                <div className="product-card__description">{description}</div>
                <h1 className="product-card__price">
                    {getPrice(price, priceSymbol)}
                </h1>
                <div
                    data-testid="product-card__category"
                    className="product-card__category"
                >
                    {category}
                </div>
            </div>
            {imgUrl && (
                <img src={imgUrl} alt={name} className="product-card__image" />
            )}
        </div>
    );
};
