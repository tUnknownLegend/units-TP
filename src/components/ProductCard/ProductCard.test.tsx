import React, { FC } from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import { ProductCard } from './ProductCard';
import { Product } from '../../types';

afterEach(jest.clearAllMocks);

jest.mock('../../utils/getPrice', () => ({
    __esModule: true,
    getPrice: jest.fn(() => '999 ₽'),
}));

const elem: Product = {
    id: 1,
    name: 'string',
    description: 'string',
    price: 1231,
    priceSymbol: '₽',
    category: 'Электроника',
    imgUrl: 'string',
};

describe('ProductCard snapshot', () => {
    it('should have img', () => {
        const renderedProductCard = render(<ProductCard {...elem} />);

        const imgElement = renderedProductCard.getByTestId(
            'product-card__image'
        );

        expect(imgElement).toBeInstanceOf(HTMLImageElement);
        expect(imgElement.getAttribute('src')).toStrictEqual(elem.imgUrl);

        expect(renderedProductCard.asFragment()).toMatchSnapshot();
    });
    it('shouldnt have img', () => {
        const renderedProductCard = render(
            <ProductCard {...elem} imgUrl={undefined} />
        );

        expect(
            renderedProductCard.queryByTestId('product-card__image')
        ).toBeNull();
        expect(renderedProductCard.asFragment()).toMatchSnapshot();
    });
});
