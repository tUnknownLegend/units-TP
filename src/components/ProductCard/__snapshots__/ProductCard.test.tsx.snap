// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductCard snapshot should have img 1`] = `
<DocumentFragment>
  <div
    class="product-card"
  >
    <div
      class="product-card__text"
    >
      <h2
        class="product-card__name"
      >
        string
      </h2>
      <div
        class="product-card__description"
      >
        string
      </div>
      <h1
        class="product-card__price"
      >
        999 ₽
      </h1>
      <div
        class="product-card__category"
        data-testid="product-card__category"
      >
        Электроника
      </div>
    </div>
    <img
      alt="string"
      class="product-card__image"
      data-testid="product-card__image"
      src="string"
    />
  </div>
</DocumentFragment>
`;

exports[`ProductCard snapshot shouldnt have img 1`] = `
<DocumentFragment>
  <div
    class="product-card"
  >
    <div
      class="product-card__text"
    >
      <h2
        class="product-card__name"
      >
        string
      </h2>
      <div
        class="product-card__description"
      >
        string
      </div>
      <h1
        class="product-card__price"
      >
        999 ₽
      </h1>
      <div
        class="product-card__category"
        data-testid="product-card__category"
      >
        Электроника
      </div>
    </div>
  </div>
</DocumentFragment>
`;
